contracts/test/TeleporterTokenDestinationTests.t.sol

```diff
@@ -64,14 +69,49 @@ abstract contract TeleporterTokenDestinationTest is TeleporterTokenBridgeTest {
         _send(input, _DEFAULT_TRANSFER_AMOUNT);
     }
 
+    function testSendAndCallingToSameInstance() public {
+        SendAndCallInput memory input = _createDefaultSendAndCallInput();
+        input.destinationBlockchainID = tokenDestination.blockchainID();
+        input.destinationBridgeAddress = address(tokenDestination);
+        _setUpExpectedDeposit(_DEFAULT_TRANSFER_AMOUNT);
+        vm.expectRevert(_formatErrorMessage("invalid destination bridge address"));
+        _sendAndCall(input, _DEFAULT_TRANSFER_AMOUNT);
+    }
+
+    function testSendZeroAmountAfterScaling() public {
+        SendAndCallInput memory input = _createDefaultSendAndCallInput();
+        input.primaryFee = 0;
+        input.secondaryFee = 0;
+        uint256 amount = 1;
+        _setUpExpectedDeposit(amount);
+        if (_scaleTokens(amount, false) == 0) {
+            vm.expectRevert(_formatErrorMessage("insufficient tokens to transfer"));
+        }
+        _sendAndCall(input, amount);
```

Comment:
I think this might make more sense as an early-exit condition at the top of the test, and then having the rest of the test as the applicable test path?